app/src/main/java/org/cru/godtools/activity/BasePlatformActivity.java

```diff
@@ -215,11 +224,17 @@ public boolean onNavigationItemSelected(@NonNull final MenuItem item) {
     protected void onStop() {
         super.onStop();
         mEventBus.unregister(this);
-        stopLanguagesChangeListener();
+        stopSettingsChangeListener();
     }
 
     /* END lifecycle */
 
+    @MainThread
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    public final void theKeyEvent(@NonNull final TheKeyEvent event) {
+        onTheKeyEvent(event);
```

Comment:
having an overridable lifecycle event is used to refresh the signup banner UI when the login state changes.